object/saml_idp.go

```diff
@@ -104,10 +106,35 @@ func NewSamlResponse(application *Application, user *User, host string, certific
 	displayName.CreateElement("saml:AttributeValue").CreateAttr("xsi:type", "xs:string").Element().SetText(user.DisplayName)
 
 	for _, item := range application.SamlAttributes {
-		role := attributes.CreateElement("saml:Attribute")
-		role.CreateAttr("Name", item.Name)
-		role.CreateAttr("NameFormat", item.NameFormat)
-		role.CreateElement("saml:AttributeValue").CreateAttr("xsi:type", "xs:string").Element().SetText(item.Value)
+		text := item.Value
+
+		// try use go-template to render it
+		tmpl, err := template.New(item.Name).Parse(item.Value)
```

Comment:
Does this affect other SAML vendors? I didn't see any if then code

Reply:
What is a "SAML vendor"? this will only affect casdoor as SAML IdP behavior, this do not related to any SAML SP behavior ( at least I cannot find out any relation )

Reply:
Does it affect other clouds? AWS, Azure, GCP? Better scope your code within a if branch, so it only affects your cloud

Reply:
没有理解SAML vendor指什么，

- 如果你在问 这是否影响casdoor作为SAML IdP对其他的下游SAML SP（i.e. 华为云 阿里云 腾讯云）： 可能会影响，如果有人用 `{{` 并且上面选择返回错误，那么旧配置会失效，签发response的时候报错；如果上面不返回错误，但之前有人按照go模板文法配置了字面值（比如之前配置成了 `{{ .user.Phone }}` ），那么assertion会改变（原先是 `{{ .user.Phone }}` 现在是 `12345678910` 这样子）
- 如果你在问 这是否影响casdoor连接上游SAML IdP的行为，这个代码和那部分没啥关系

Reply:
> Does it affect other clouds? AWS, Azure, GCP? Better scope your code within a if branch, so it only affects your cloud. I mean it should not fix your cloud's bug while introducing new bugs for other clouds

This is really a generic solution, one can use this template syntax to map any assertion for any SP, even with some if-else logic in template `{{ if host eq "some.sp.com" }}A{{else}}B{{ endif }}`

这是个很通用的解决方案：如果你不需要模板，那么直接字面值就行，如果你需要，这个模板适用于任何SP，你甚至可以在模板里写判断 `{{ if host eq "some.sp.com" }}A{{else}}B{{ endif }}`